specs/execution-api.md

```diff
@@ -1,128 +1,150 @@
-in a rollups genesis, and used by the conductor as configuration on startup.
+`CommitmentState` contains the field `lowest_celestia_search_height`, representing
+the lowest Celestia height that will be searched for the next firm block. In the
+current implementation of Conductor, this is set to the Celestia height at which
+the most recent firm block was found.
 
-### ExecuteBlock
+There are, however, many factors that can result in the same Sequencer blocks sharing
+or skipping Celestia heights. As such, the Celestia heights correlating to Sequencer
+commitments may not increase linearly, and a range of heights must be searched.
+Conductor begins fetching at `CommitmentState.lowest_celestia_search_height`, continuing
+searching for firm commitments until it reaches `lowest_celestia_search_height +
+ExecutionSessionParameters.celestia_search_height_max_look_ahead`.
 
-`ExecuteBlock` executes a set of given transactions on top of the chain
-indicated by `prev_block_hash`. The following should be respected:
+## Rollup Implementation Details
 
-- `prev_block_hash` MUST match hash of the `SOFT` commitment state block, return
-  `FAILED_PRECONDITION` otherwise.
-- If block headers have timestamps, created block MUST have matching timestamp
-- The CommitmentState is NOT modified by the execution of the block.
+### CreateExecutionSession
 
-### GetBlock
+`CreateExecutionSession` returns an `ExecutionSession`, defining all necessary information
+for the client to begin driving execution. This includes `ExecutionSessionParameters`,
+containing the necessary information for network connection and mapping rollup blocks
+to Astria Sequencer (soft) and Celestia (firm) blocks. Also returned is the current
+`CommitmentState` of the server and a `session_id` to be supplied in RPC requests
+for the current session.
 
-`GetBlock` returns information about a block given either its `number` or
-`hash`. If the block cannot be found return a `NOT_FOUND` error.
+This RPC should **always** succeed. The API is agnostic as
+to how this information is defined in a rollup's genesis, and is only used by the
+Conductor as configuration at the beginning of each execution session.
 
-### BatchGetBlocks
+### ExecuteBlock
 
-`BatchGetBlocks` returns an array of Blocks which match the array of passed in
-block identifiers.
+`ExecuteBlock` executes a set of given transactions on top of the chain
+indicated by `prev_block_hash`. The following should be respected:
 
-- The API endpoint MUST fail atomically, returning either all requested resources
-  or a `NOT_FOUND` error.
-- The returned objects MUST be in the same order as they were requested.
+- `prev_block_hash` MUST match hash of the most recently committed block, whether
+  `soft` or `firm`. RPC should return `FAILED_PRECONDITION` status otherwise.
+- If block headers have timestamps, the created block MUST have matching timestamp
+- **NOTE:** The `CommitmentState` is NOT modified by the execution of the block.
```

Comment:
Need to update to add information on the `sequencer_block_hash` which has since been added here